src/util/aws-util.ts

```diff
@@ -81,28 +82,25 @@ export class AwsUtil {
      *
      * If profile is provided, then partition will be ignored.
      */
-    public static async Initialize(): Promise<void> {
-        let stsClient = new STSClient();
-        // running with a profile in the commercial region
-        if (this.profile) {
-            process.env.AWS_SDK_LOAD_CONFIG = '1';
-            process.env.AWS_PROFILE = this.profile;
-            const caller = await stsClient.send(new GetCallerIdentityCommand({}));
-            this.userId = caller.UserId.replace(':', '-').substring(0, 60);
-            return;
-        }
-        // targeting a partition
-        if (this.isPartition) {
-            // if targeting partition with a profile
-            if (this.partitionProfile) {
-                process.env.AWS_SDK_LOAD_CONFIG = '1';
-                process.env.AWS_PROFILE = this.partitionProfile;
-            }
-            stsClient = new STSClient({ region: this.partitionRegion });
-        }
+    public static async Initialize(credentials?: Provider<AwsCredentialIdentity>): Promise<void> {
+
+        // if no credentials(provider) was passed in this method, we use the default
+        const provider = credentials ?? defaultProvider({
```

Comment:
is this custom credentials provider strictly here for testing purposes? I can't find anywhere in the code where it is invoked with a customer credentials provider

Reply:
we used to do that for GovCloud (see [here](https://github.com/org-formation/org-formation-cli/blob/416dbaaf6f531a76278d8226850336a4398eec58/src/util/aws-util.ts#L96C12-L101C16))